use of undeclared crate or module `futures` with polars 0.43.0

### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

Enable the following features in `Cargo.toml`:

```toml
[dependencies]
polars = { version = "0.43.0", features = ["lazy", "parquet", "streaming"] }
```

And try to compile any program.

### Log output

```shell
Compiling polars-pipe v0.43.0
error[E0432]: unresolved import `futures`
 --> /Users/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/polars-pipe-0.43.0/src/executors/sources/parquet.rs:7:5
  |
7 | use futures::{StreamExt, TryStreamExt};
  |     ^^^^^^^ use of undeclared crate or module `futures`

For more information about this error, try `rustc --explain E0432`.
```


### Issue description

It looks like the `futures` feature needs to be enabled with this combination of features.

### Expected behavior

No error

### Installed versions

See above

--- a/crates/polars-pipe/Cargo.toml
+++ b/crates/polars-pipe/Cargo.toml
@@ -35,7 +35,7 @@ version_check = { workspace = true }
 [features]
 csv = ["polars-plan/csv", "polars-io/csv"]
 cloud = ["async", "polars-io/cloud", "polars-plan/cloud", "tokio", "futures"]
-parquet = ["polars-plan/parquet", "polars-io/parquet", "polars-io/async"]
+parquet = ["polars-plan/parquet", "polars-io/parquet", "polars-io/async", "futures"]
 ipc = ["polars-plan/ipc", "polars-io/ipc"]
 json = ["polars-plan/json", "polars-io/json"]
 async = ["polars-plan/async", "polars-io/async", "futures"]

